rocketpy/environment/environment.py

```diff
@@ -3029,34 +2328,45 @@ def pressure_function(h):
             layer = bisect.bisect(geopotential_height, H) - 1
 
             # Retrieve layer base geopotential height, temp, beta and pressure
-            Hb = geopotential_height[layer]
-            Tb = temperature[layer]
-            Pb = pressure[layer]
+            base_geopotential_height = geopotential_height[layer]
+            base_temperature = temperature[layer]
+            base_pressure = pressure[layer]
             B = beta[layer]
 
             # Compute pressure
             if B != 0:
-                P = Pb * (1 + (B / Tb) * (H - Hb)) ** (-g / (B * R))
+                P = base_pressure * (
+                    1 + (B / base_temperature) * (H - base_geopotential_height)
+                ) ** (-g / (B * R))
             else:
-                T = Tb + B * (H - Hb)
-                P = Pb * np.exp(-(H - Hb) * (g / (R * T)))
-
-            # Return answer
+                T = base_temperature + B * (H - base_geopotential_height)
+                P = base_pressure * np.exp(
+                    -(H - base_geopotential_height) * (g / (R * T))
+                )
             return P
 
-        # Save international standard atmosphere pressure profile
-        self.pressure_ISA = Function(
-            pressure_function,
-            inputs="Height Above Sea Level (m)",
-            outputs="Pressure (Pa)",
-        )
-
-        # Discretize Function to speed up the trajectory simulation.
-        self.barometric_height_ISA = self.pressure_ISA.inverse_function().set_discrete(
-            pressure[-1], pressure[0], 100, extrapolation="constant"
-        )
-        self.barometric_height_ISA.set_inputs("Pressure (Pa)")
-        self.barometric_height_ISA.set_outputs("Height Above Sea Level (m)")
+        # Discretize this Function to speed up the trajectory simulation
+        altitudes = np.linspace(0, 80000, 100)  # TODO: should be -2k instead of 0
+        pressures = [pressure_function(h) for h in altitudes]
+
+        return np.column_stack([altitudes, pressures])
+
+    @funcify_method("Pressure (Pa)", "Height Above Sea Level (m)")
+    def barometric_height_ISA(self):
+        """Returns the inverse function of the pressure_ISA function."""
+        return self.pressure_ISA.inverse_function()
+
```

Comment:
This is not setting discrete like in the previous code. Why?

Reply:
The `barometric_height_ISA` is already discretized, therefore the set_discretize() call was not necessary in here.